tools/mapper/find_proxy_function.py

```diff
@@ -0,0 +1,252 @@
+        log.warning(f"Correlation cannot be found. Error: {e}")
+        return math.inf
+
+    # NOTE: Better Error mechanisms can be considered to improve the correlation
+    # Currently only the 75%tile memory and duration are considered.
+    # Euclidean distance between normalized memory and duration is considered
+    try:
+        if trace_memory == 0: trace_memory += 0.01
+        if trace_duration == 0: trace_duration += 0.01
+        diff_memory = (math.log(trace_memory) - math.log(proxy_memory)) 
+        diff_duration = (math.log(trace_duration) - math.log(proxy_duration)) 
+        error = math.sqrt((diff_memory) ** 2 + (diff_duration) ** 2)
+        return error
+    except ValueError as e:
+        log.warning(f"Correlation cannot be found. Error: {e}")
+        return math.inf
+
+
+def get_proxy_function_using_linear_sum_assignment(
+    trace_functions: dict, proxy_functions: dict
+) -> Tuple[dict, int]:
+    """
+    Obtains the one-to-one mapped proxy function for every trace function
+    
+    Parameters:
+    - `trace_functions` (dict): Dictionary containing information regarding trace functions
+    - `proxy_functions` (dict): Dictionary containing information regarding proxy functions
+    
+    Returns:
+    - `dict`: Dictionary containing information regarding trace functions with the associated proxy functions
+    - `int`: 0 if no error. -1 if error
+    """
+
+    try:
+
+        trace_functions = OrderedDict(trace_functions)
+        proxy_functions = OrderedDict(proxy_functions)
+
+        trace_list = []
+        for tf in trace_functions:
+            trace_list.append(trace_functions[tf])
+            trace_functions[tf]["index"] = len(trace_list) - 1
+
+        proxy_list = []
+        for pf in proxy_functions:
+            proxy_list.append(proxy_functions[pf])
+            proxy_functions[pf]["index"] = len(proxy_list) - 1
+
+        # Creating error matrix
+        m, n = len(trace_functions.keys()), len(proxy_functions.keys())
+        error_matrix = np.empty((m, n))
+
+        # This utilized Jonker-Volgenant algorithm for Linear Sum assignment - scipy package
+        # to calculate the best possible assignment for the trace functions
+        # Time complexity : O(n^3) where n is the largest of number of rows/columns
+        for i in range(m):
+            for j in range(n):
+                error_matrix[i, j] = get_error(trace_list[i], proxy_list[j])
+
+        # Do the linear sum assignment problem
+        row_indices, col_indices = sp.linear_sum_assignment(error_matrix)
+        assignments = list(zip(row_indices, col_indices))
+
+        # Go through the assignment solution
+        for assignment in assignments:
+            row_index = assignment[0]
+            col_index = assignment[1]
+            trace = ""
+            proxy = ""
+            for tf in trace_functions:
+                if row_index == trace_functions[tf]["index"]:
+                    trace = tf
+                    break
```

Comment:
You linearly search in the dict. Are sure that this data structure is suitable for the task? You also literally created trace_list that has all of them in the index order.